### PR TITLE
Generate Resources SDK with latest version of swagger spec. 

### PR DESCRIPTION
**Pull request by TianoMS**

### CODE REVIEW

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/generate.cmd [hunk 1]
```diff
@@ -9,7 +9,7 @@ set autoRestVersion=0.17.0-Nightly20160629
 set source=-Source https://www.myget.org/F/autorest/api/v2
 
 set resSpecFile="https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-resources/resources/2016-09-01/swagger/resources.json"
-set lockSpecFile="https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-resources/locks/2015-01-01/swagger/locks.json"
+set lockSpecFile="https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/arm-resources/locks/2016-09-01/swagger/locks.json"
```
Comment (cormacpayne) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/generate.cmd]: @TianoMS please update the `autoRestVersion` on line 8 with the nightly build version you used

File: src/ResourceManagement/Resource/Resource.Tests/InMemoryTests/DeploymentTests.InMemory.cs [hunk 2]
```diff
@@ -735,23 +735,6 @@ public void DeploymentTestsCancelThrowsExceptions()
         }
 
         [Fact]
-        public void DeploymentTestsWithoutModeThrowsExceptions()
```
Comment (cormacpayne) [inline src/ResourceManagement/Resource/Resource.Tests/InMemoryTests/DeploymentTests.InMemory.cs]: @TianoMS why was this test deleted?

**Review by cormacpayne — DISMISSED** [review]

File: src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/generate.cmd (hunk 1 above)
Comment (TianoMS) [inline src/ResourceManagement/Resource/Microsoft.Azure.Management.ResourceManager/generate.cmd]: I was using a private package provided by Garrett. He has some fixes there. I think I can update the version later when he has all the fixes checked in.

File: src/ResourceManagement/Resource/Resource.Tests/InMemoryTests/DeploymentTests.InMemory.cs (hunk 2 above)
Comment (TianoMS) [inline src/ResourceManagement/Resource/Resource.Tests/InMemoryTests/DeploymentTests.InMemory.cs]: There were some changes in AutoRest and now the Enum types are no longer generated as nullable types. The validation on null values was also removed. As a result, this test doesn't apply now.

**Comment by markcowl** [comment]
@TianoMS We are working out how to properly archive this version of the generator so you can refer to it.  We'll suggest a minor change to the generation file after this.

**Comment by TianoMS** [comment]
@markcowl Yes I'll regenerate the files after I get a AutoRest package with the fixes.

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by shahabhijeet** [comment]
@TianoMS please change the title when you update the PR with the proposed set of changes.

**Pull request merged**